docs: Apply editor configurations from a ConfigMap (#2841)
codenvy/che-docs: modules/administration-guide/pages/editor-configurations-for-microsoft-visual-studio-code.adoc

:_content-type: PROCEDURE
:description: Applying editor configurations
:keywords: settings, extensions, configurations
:navtitle: Applying editor configurations
// :page-aliases:

[id="visual-studio-code-editor-configs"]
= Applying editor configurations

You can configure Visual Studio Code - Open Source editor by adding configurations to a ConfigMap.
These configurations are applied to any workspace you open.
Once a workspace is started, the editor checks for this ConfigMap and stores configurations to the corresponding config files.

The following sections are currently supported:

* settings.json
* extensions.json

> `settings.json` section contains various settings that allow to customize different parts of the Code - OSS editor. +

Reviewer: ```suggestion
The `settings.json` section contains various settings that allow to customize different parts of the Code - OSS editor. +
```

> `extensions.json` section contains recommented extensions are installed when a workspace is started.

Reviewer: ```suggestion
The `extensions.json` section contains recommented extensions are installed when a workspace is started.
```

.Procedure

> * Add a new ConfigMap to the user's {orch-namespace}, define the __settings.json__  and __extensions.json__ sections, specify the settings you want to add and the IDs of the extensions you want to install.

Reviewer: ```suggestion
* Add a new ConfigMap to the user's {orch-namespace}, define the `__settings.json__`  and `__extensions.json__` sections, specify the settings you want to add, and the IDs of the extensions you want to install.
```

+
====
[source,yaml]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: vscode-editor-configurations
data: 
  extensions.json: |
    {
      "recommendations": [
          "dbaeumer.vscode-eslint",
          "github.vscode-pull-request-github"
      ]
    }
  settings.json: |
    {
      "window.header": "A HEADER MESSAGE",
      "window.commandCenter": false,
      "workbench.colorCustomizations": {
        "titleBar.activeBackground": "#CCA700",
        "titleBar.activeForeground": "#ffffff"
      }
    }
immutable: false
----
====

* Start or restart your workspace 

[NOTE]
====

> Ensure the Configmap contains data in a valid JSON format.

Reviewer: ```suggestion
Make sure that the Configmap contains data in a valid JSON format.
```
